src/spec-common/dotfiles.ts

```diff
@@ -45,14 +45,14 @@ export async function installDotfiles(params: ResolverParameters, properties: Co
 ${createFileCommand(markerFile)} || (echo dotfiles marker found && exit 1) || exit 0
 command -v git >/dev/null 2>&1 || (echo git not found && exit 1) || exit 0
 [ -e ${targetPath} ] || ${env}git clone ${repository} ${targetPath} || exit $?
-if [ -x ${installCommand} ]
+cd ${targetPath}
+if [ ! -x ${installCommand} ]
 then
-	echo Executing script ${installCommand}
-	cd ${targetPath} && ${env}${installCommand}
-else
-	echo Error: ${installCommand} not executable
-	exit 126
+	echo Setting '${targetPath}'/'${installCommand}' as executable
+	chmod +x ${installCommand}
 fi
+echo Executing script ${installCommand}
+${env}./${installCommand}
```

Comment:
Curious, why did we change `${env}${installCommand}` to `${env}./${installCommand}` ?

Reply:
I changed it here to align with the other block.  Functionally it should be no different

Reply:
This would break cases, e.g., that use `~/targetPath/installScript.sh`. Should we keep the previous variant to minimize the risk of breaking existing configurations? (Below we know that `$installCommand` is a filename without path component.)

Reply:
It appears this flag had been broken unless you provided it an absolute path. 
<img width="1800" alt="image" src="https://github.com/devcontainers/cli/assets/23246594/e02473c5-4e08-4a69-9495-2151d844c31b">

I've made changes to support both just a file name in the root of the dotfiles directory, as well as an absolute path (if the former doesn't resolve as a file).

I added several new tests to confirm.